### DISCUSSION TITLE
New iOS app: sync fails

### DISCUSSION THREAD

**Discussion by aksdb** [dani-garcia/vaultwarden#4921] [Q&A]
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.32.0
* Web-vault version: v2024.6.2b
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Alpine)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: false (X-Forwarded-For)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: false
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.46.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** SIGNUPS_ALLOWED, ADMIN_TOKEN

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_max_note_size": 10000,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://***********",
  "domain_origin": "*****://***********",
  "domain_path": "",
  "domain_set": true,
  "duo_context_purge_schedule": "30 * * * * *",
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "duo_use_iframe": false,
  "email_2fa_auto_fallback": false,
  "email_2fa_enforce_on_verified_invite": false,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "enable_websocket": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "http_request_block_non_global_ips": true,
  "http_request_block_regex": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "increase_note_size_limit": false,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Bitwarden_RS",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "push_enabled": false,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": true,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": false,
  "smtp_from": "",
  "smtp_from_name": "Bitwarden_RS",
  "smtp_host": null,
  "smtp_password": null,
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": true,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

### Vaultwarden Build Version

v1.32.0

### Deployment method

Official Container Image

### Reverse Proxy

caddy 2.8.4

### Host/Server Operating System

Linux

### Clients

iOS

### Client Version

2024.8.1 (1052)

### Steps To Reproduce

Open the iOS app.

### Expected Result

See a list of vault entries.

### Actual Result

"An error has occured"

### Additional Context

Unfortunately the app doesn't provide any more details, so it might as well be an actual problem of the app. But since I use vaultwarden vs the official server, I don't want to open a bug report with them.

**Comment by BlackDex** [comment]
What are the logs from Vaultwarden?
We need more details.

**Comment by gerroon** [comment]
I have the same issue. this is what I get when I open the app that fails. I cant tell if the app is at fault or the server is. Everything was working fine untile their last update this morning for me

Vaultwarden Web
Version 2023.10.0  (Docker(

```
[2024-09-04 16:17:44.017][request][INFO] GET /api/sync
[2024-09-04 16:17:44.023][request][INFO] GET /api/sync
[2024-09-04 16:17:46.534][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
[2024-09-04 16:17:46.540][response][INFO] (sync) GET /api/sync?<data..> => 200 OK

```

EDIT: I just did a docker pull and the app works now. So I am not sure what the issue is actually.

**Reply by aksdb** [reply]
Same logs here, when opening the app:

```
[2024-09-04 17:17:05.677][request][INFO] GET /api/sync
[2024-09-04 17:17:05.729][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
```

**Comment by BlackDex** [comment]
That is strange. Cache maybe?

**Reply by aksdb** [reply]
I tried to log out and in again (which I guess should eliminate the cache?), but the same error occured. Which is why I narrowed it down to sync. The other calls seem to be fine. If I had to guess, I would assume it expects a slightly different response.

(Or, as I said, it could be something completely different and indeed a bug in the iOS app.)

**Reply by aksdb** [reply]
Also reinstalled the app now, so there should have been no left-overs. Still the same error (after logging in).

**Reply by winny1972** [reply]
I have the same problem with ios app and self-hosted server after last app update

**Reply by msiemers** [reply]
Here the same Problem.

**Reply by halms** [reply]
I also got the same problem, mentioned it in the [issue](https://github.com/dani-garcia/vaultwarden/issues/4870#issuecomment-2329689825) about the Android app.

What I've tested so far:
- It's not 2FA
- It works fine with a new, empty account.

So I'd agree, it's probably a slight difference in the response of the server that the new client can't parse.

**Reply by BlackDex** [reply]
Please read https://github.com/dani-garcia/vaultwarden/issues/4870#issuecomment-2306749481 and if you want send the data please do so that we might be able to fix this.

**Reply by halms** [reply]
Is it enough if I send the response to the `/sync` request? Since this appears to be the problematic one here?

**Reply by BlackDex** [reply]
@halms If you can see the vault interface with an error and the sync call succeeded with an ok on the Vaultwarden side. Then yes, that would be enough. Again, all is encrypted, but the full output is needed.

**Reply by Gerardv514** [reply]
> Please read [#4870 (comment)](https://github.com/dani-garcia/vaultwarden/issues/4870#issuecomment-2306749481) and if you want send the data please do so that we might be able to fix this.

Why close issue #4920  and turn this into a discussion? Seems like multiple people reporting an issue. I for one haven’t updated my iOS app until I see how this pans out.

**Reply by BlackDex** [reply]
Since if it this is an issue with the response of Vaultwarden then there already is an item open. That does have android in the title, but the case is the same. 

Keeping two issues open with the same story is not needed.

**Reply by halms** [reply]
Okay, good news!
I was able to solve it for my case.
Details here: https://github.com/dani-garcia/vaultwarden/issues/4870#issuecomment-2330302849

**Comment by Gerardv514** [comment]
Does anyone know of a way to revert the iOS app if there is an issue?

I can update the app to see if I experience similar issues, but not if I can revert.

**Reply by BlackDex** [reply]
Only way i know of is via itunes backup. But that was a long time ago.

**Reply by Gerardv514** [reply]
Let me try a secondary device such as an iPad.

**Reply by Gerardv514** [reply]
No apparent issues on the iPad with new app

**Comment by halms** [comment]
I was able to solve this issue for my case, details in the issue: https://github.com/dani-garcia/vaultwarden/issues/4870#issuecomment-2330302849

**Comment by cloudguy11** [comment]
It seems there are different problems by different users regarding Bitwarden IoS 2024.8.1 errors. App only shows "An errors has occured". 

My setup:  Tried both Vaultwarden and Bitwarden open source as Docker on Synology NAS. Synology DDNS and reverse proxy with Let's encrypt certificate. DNS entry points to intranet 192.168.178.x ( I use all clients only within the intranet).
Both approaches work with https browser access, and also with a Bitwarden app on MacOS, but not with the IpadOS app 2024.8.1.
I followed these instructions: https://drfrankenstein.co.uk/vaultwarden-bitwarden-in-container-manager-on-a-synology-nas/ 
with the exception of not exposing to internet, but using only intranet (local Wifi) with a private IP-adress resolved by DNS.

Findings:

- Bitwarden MacOs App works flawless with this setup, both for Vaultwarden and Bitwarden Server.
- IoS 18 Public Beta Bitwarden client 2024.8.1 does not work with my setup, neither Vaultwarden-Server nor Bitwarden-Server. Haven't tried any previous versions, this is my first setup.
- Tcpdump showed connections behind the reverse proxy for Bitwarden MacOs app, but nothing for the Bitwarden IoS App, indicating that the IoS Apps don't even get beyond the reverse proxy.

Summary:
Don't think it's a Vaultwarden issue, in my specific case. The new 2024.8.1 Bitwarden app, doesn't even want to pass beyond the Synology NAS reverse proxy (with it's Let's Encrypt cert). 
My next step would have been to NOT use the Synology reverse proxy and to try a separate, real local nginx or similar setup. If anyone has an advise, or had a simililar problem in the past, that would be appreciated.

**Reply by Gerardv514** [reply]
I am using a synology nas with vaultwarden in a docker container. I am using built in reverse proxy, I am not seeing any issues with the latest iOS app rolled out. Additionally I am running latest VW image (not testing).

**Reply by cloudguy11** [reply]
Thank you, that's interesting. Still doesn't work for me. Do you have any online instructions which you followed, which I can then try to repeat ? I am new, hence if this is not the right place, to ask such questions, let me know please.

My docker-compose ( tried both server:testing and server:latest)

<img width="656" alt="image" src="https://github.com/user-attachments/assets/4a99a04a-7c39-42c4-8310-d4b33add4dc3">


My temporary DDNS setup with Let'S Encrypt cert resolving to an intranet addresss: 192.168.178.86 ( clients only used in local wifi network).
<img width="750" alt="image" src="https://github.com/user-attachments/assets/439af4a1-b217-493f-a182-abedf4466822">

Reverse Proxy:
<img width="588" alt="image" src="https://github.com/user-attachments/assets/9c729fb8-930b-43ef-87ab-67e51d8397a9">

<img width="588" alt="image" src="https://github.com/user-attachments/assets/3e936878-53fc-4131-b92f-12a136bead9a">

**Reply by Gerardv514** [reply]
I've used this resource. Looking at your port, 8080 doesn't appear correct. Should be port 80. Rocket port is 8080 so there maybe a conflict there.

https://www.synoforum.com/resources/bitwarden-self-hosted-password-manager-using-vaultwarden-server-image.3/

**Reply by Gerardv514** [reply]
Curious, did you add an allow rule into your firewall? If so what was the port you used in the firewall entry?

**Reply by cloudguy11** [reply]
I was able to solve the Problem. In the end it was trivial, but difficult to find, and I only sorted it out after quite some time:

My working Setup:
- Vaultwarden as Docker on Synology in Intranet: 192.168.x.x
- Custom Synology Nginx configuration (I guess that the Synolgoy standard reverse proxy UI config will do as well)
- DDNS resolving to intranet: 192.168.x.x only (note: All clients only syncing with server when being at home)
- Manual DNS provider config on all devices (e.g. google DNS: 8.8.8.8 and 8.8.4.4)

Previous Problem Analysis: Tcpdump packet trace on IPhone showed a DNS lookup attempt for my vaultwarden Synology DDNS Domain. However the DNS-server response didn't show a resolved IP-address, failed !
Packet trace instructions: https://developer.apple.com/documentation/network/recording-a-packet-trace

My Mac was able to connect to Vaultwarden any time, including the Bitwarden Mac client. I only figured out lately that the Mac had configured a manual DNS ( e.g. 8.8.8.8) and my mobile devices were using the default Wifi-router/internet-provider configured DNS.  Hence mobile devices were not able to resolve my vaultwarden synology DDNS domain, while being able to resolve any other domain though.

Simply configuring a manual DNS of e.g. 8.8.8.8 in the iPhone settings solved the issue right away.

Trivial, but difficult to find. I had suspected a SW configuration issue and tried several different bittwarden and vaultwarden-docker variants with and without separate nginx setups on Synology, to no avail.

Advise for other starters:

1. Security: You can setup a DDNS pointing to intranet only, e.g. 192.168.x.x., while still allowing you to have a required SSL certificate for a (non publicly reachable) server. Hence your self-hosted vaultwarden server is in intranet (home wifi) only, and not attackable from internet. Your clients, which you sync with the service when being at home, will anyway cache the data locally, so us can 'access/read' the passwords when you're on the move.

2. If Bitwarden clients don't work on iPhone/iPad/Android, but work on a Laptop device:  As a first step try to open your vaultwarden-server web-page also on your mobile device, to see if it's a DNS resolution or connection issue first.

3. If the DNS isn't the issue, then taking a network trace from your iPhone/Ipad using a MacBook and a "Remote Virtual Interface" may help for furher analysis: https://developer.apple.com/documentation/network/recording-a-packet-trace

4. Certain access-points/routers, for example AVM FritzBox do not support DNS resolution that point to a private Ip-address in the same home network: https://en.avm.de/service/knowledge-base/dok/FRITZ-Box-7360-int/663_No-DNS-resolution-of-private-IP-addresses/

**Comment by AndBae** [comment]
Vaultwarden 1.32.0 Docker container on Synology NAS with local reverse proxy and Let's Encrypt certificate.
Bitwarden 2024.8.1 (1052) iOS application installed today.
Everything works perfectly on iPhone and iPad Pro.

**Comment by 11jwolfe2** [comment]
I upgraded to the new iOS app 2024.8.1 and moved a password from my personal vault my my family organization and now I get the following error on my phone. "Cryptograpy error. The cipher's MAC doesn't match the expected value". So nothing loads on my phone. When I go to the WEB UI it spins and loads really slow but eventually loads my vault, I however can not find the password I moved. I am guessing something got corrupted but im not 100% sure. 

Any ideas? 

I have updated my vaultwaredn to the testing build. 
I have uninstalled and reinstalled the app several times logged in and logged out. 
Nothing odd in the logs. I saw something that prompted me to add web socket support to my nginx proxy but still either a slow AF sin on web or none at all on mobile.

**Reply by Gerardv514** [reply]
This has been reported already and fixed in the latest version. Unfortunately once the corruption happens you need to do something in the database to fix. 

Reference #4767 and continue discussion there to keep the issues organized.

**Reply by 11jwolfe2** [reply]
Thank you I did not see that discussion!

**Comment by aksdb** [comment]
@BlackDex I set up a local debug environment (only the network side; I don't have a Mac to actually debug the iOS app and the Bitwarden devs decided to send debug logs to crashlytics instead of the device log where I could have intercepted it).

Anyway, I am not yet sure if it's the only problem, but _one_ problem at least is the format of secure notes.

Old notes in my vault have a structure like this (only the relevant parts):

```json
{
  "type": 2,
   "data": {
      "name": "...",
       "fields": [...]
    },
    "fields": [...],
    "secureNote": {
      "name": "...",
      "fields": [...]
    }
}
```

New notes on the other hand look like this:

```json
{
  "type": 2,
   "data": {
      "name": "...",
       "fields": [...]
    },
    "fields": [...],
    "secureNote": {
      "type": 0
    }
}
```

As you can see, the "secureNote" field does apparently no longer contain (a third?!) copy of the data, but simply states `"type": 0`.

With the old structure, my sync fails. With the content of the `secureNote` field replaced, the sync succeeds and the note shows up.

Edit: I just realized: it doesn't even care about the other values in `secureNote`, but the _missing_ `"type": 0` in there causes a problem. So I can even fix my old notes by just added `"type": 0` to the `secureNote` object.

**Reply by aksdb** [reply]
Ah, it looks like currently vaultwarden "only" fixes cases, where `data` is basically empty:

```
        if self.atype == 2
            && (self.data.is_empty() || self.data.eq("{}") || self.data.to_ascii_lowercase().eq("{\"type\":null}"))
        {
            type_data_json = json!({"type": 0});
        }
```

When old entries, like in my case, have some data set, but it doesn't contain `type`, then the current check would obviously not hit.

Urgh, if the structure needs to be merged, we would have to actually parse and modify the JSON then, I guess. On the other hand ... are there actually cases where `secureNote` should contain anything besides the type? I guess I need to dive into bitwarden's client and server code a bit ...

**Reply by aksdb** [reply]
Unless I miss something, it should indeed always only have a field named `type`: https://github.com/bitwarden/server/blob/main/src/Api/Vault/Models/CipherSecureNoteModel.cs

And so far the only valid type is `0`: https://github.com/bitwarden/server/blob/main/src/Core/Vault/Enums/SecureNoteType.cs

**Reply by aksdb** [reply]
I modified the vaultwarden code above to naively just return `"type": 0`:
```rust
        if self.atype == 2
        {
            type_data_json = json!({"type": 0});
        }
```
Now the sync works for me.

I am not sure this fix is a good idea, since theoretically they could introduce a new type. If we want to behave like the official server, we would have to map the `type` from `data` and default it to 0 if it's not set, I guess.
Optionally it would likely also work to do what I proposed above to keep `data` unmodified and add the field `type` if it's missing. But I wouldn't want to do this as a string operation like is currently done.

**Reply by aksdb** [reply]
(Sorry for thinking out loud here. Just documenting my thoughts)

Actually now I wonder why the fix above currently even is done with a check on the raw data instead of the already parsed `type_data_json` :thinking: .

**Reply by aksdb** [reply]
So at least for me this works (`cipher.rs:192`):
```rust
        if self.atype == 2 {
            if type_data_json["type"].is_null() {
                type_data_json["type"] = json!(0);
            }
        }
```
That also aligns better with the other fixes in this file.

Tomorrow I'll dive deeper if there are tests somewhere I need to adjust. Then I would do so and open a PR.

**Reply by BlackDex** [reply]
I already have this on a draft PR #4932 
Need to do some other fixes to.

**Reply by philpennock** [reply]
I too am seeing this issue with the iOS client after upgrading, and I built a new vaultwarden server with #4932 applied, `1.32.0-22-g2d2e72df` as the repo status.
It's not yet fixed for me.  Logs from the server-side are just:

```
[2024-09-08 08:03:54.929][request][INFO] GET /api/sync
[2024-09-08 08:03:55.328][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
```

Android app, iOS client on iPad (not upgraded), browser extensions, Linux v 2024.8.2 GUI app and CLI `bw sync`, are all happy with that vaultwarden build.

**Reply by BlackDex** [reply]
That it is but fixed isn't strange as the PR is not yet done and thus not yet merged.

**Reply by aksdb** [reply]
@philpennock are you up for debugging? If you want to fiddle with the client side, you could probably build the Android app in debug mode and get actual error messages out of it.

If you want to fiddle with only the server side, you could use `mitmproxy` between your normal reverse-proxy and vaultwarden (so reverseproxy points at mitmproxy which points at vaultwarden) then you can intercept the responses and also fake responses. That's how I did the debugging I mentioned above. I stripped out parts of the sync data until the sync succeeded. Then I put data in smaller chunks back in until it failed again. Then I knew which entry to look at.

**Reply by philpennock** [reply]
Sorry, been away from my computer for a few days.  I knew the PR wasn't complete yet, was hoping for a better way to get diagnostics from the client or enable debug logging in the vaultwarden server to have more details to be more helpful in a follow-up.  I haven't built an Android app in a decade and that app isn't broken (only the iOS rewrite) so I don't think that would help? 
 My server deployment is in K3S, so it would be quite involved to inject something else into it, so is a last resort if there's really no debug logging I can enable.

Have rebuilt on merged main at `1.32.0-23-g25d99e35` to include the non-WIP version of the patch.  Still not fixed for me.

I will try to sort out mitmproxy-in-k8s deployment this weekend.  Thanks.

**Reply by aksdb** [reply]
@philpennock you can also spin up vaultwarden with a copy of your DB locally and connect the bitwarden client through your local network (just via http and IP address to your desktop). That should suffice to reproduce the problem and then narrow it down.

The server itself can't help you with logs, since it's the client that fails to parse the response. The client should be able to log something, but the non-debug builds all send their details to crashlytics and only show the user this dumb non-helping error message. I hope they improve that at some point, but there is nothing vaultwarden can do about it, unfortunately.

**Reply by BlackDex** [reply]
What you need is the sync response from Vaultwarden.
That can come from a browser too. I wonder how your secure note records look like.

@aksdb can you provide what is actually in database for those secure notes in the data column? I have seen a lot of weird combos there, but never the records you mention there like name and fields. 

Because that i do not filter/check. If those also contain a type field, then it will now be kept.

**Reply by aksdb** [reply]
@BlackDex here is an example of one of the notes (redacted):
```json
{
    "Favorite": false,
    "Fields":
    [
        {
            "Name": "....",
            "Type": 1,
            "Value": "...."
        }
    ],
    "Name": "....",
    "Notes": null,
    "RevisionDate": "2018-11-01T16:57:02Z"
}
```

**Reply by BlackDex** [reply]
That looks more like a login item i think

**Reply by aksdb** [reply]
It has `atype == 2`. I store stuff like software license codes in secure notes where the actual payload is in custom fields, which explains why tehre are fields but no actual note. I basically use secure notes as "custom entries" (because they are the most bare bone bitwarden gives me).

```sql
SELECT created_at, data->'Type', data FROM ciphers
    WHERE atype==2 and data->'Type' is null;
```

Also I noticed that all the affected notes are from the end of 2018 (november and december), so from around vaultwarden 1.0 -and whatever version the bitwarden clients had back then.

---

Additional finding: Running that query I realized, that the current iOS bitwarden client seems to write `"type": 0`, while other/older versions write `"Type": 0` ... :roll_eyes: ... so I guess some normalization of the keys might be in order as well.

```sql
SELECT created_at, atype, data FROM ciphers
    WHERE atype==2 and (data->'Type' is not null or data->'type' is not null) and created_at >= '2024-03-06 18:27:26.282451204';
```

```json
[
  {
    "created_at": "2024-03-06 18:27:26.282451204",
    "atype": 2,
    "data": "{\"Type\":0}"
  },
  {
    "created_at": "2024-09-07 16:12:21.507640115",
    "atype": 2,
    "data": "{\"type\":0}"
  },
  {
    "created_at": "2024-09-07 16:49:14.273893165",
    "atype": 2,
    "data": "{\"type\":0}"
  }
]
```

**Reply by BlackDex** [reply]
Yes, the new clients use camelCase instead of PascalCase. Vaultwarden is already adjusted to return all items in camelCase and PascalCase where still needed otherwise.

The old clients accept both, while the new only accepts camelCase (in most places), and is case sensitive.

This all is already fixed and should be working in the `testing` tagged images out of the box!

**Comment by jack228cn** [comment]
I had to backup the data and rebuild the docker with latest version, it worked.

**Comment by XRTce** [comment]
I had the same issue. Nothing mentioned here worked. I finally found the issue: i had 3 entries with no name. They were showing in the web-ui as blank row. I went ahead and deleted the entries and the mobile app is working now

**Reply by Gerardv514** [reply]
Were these entries that you created? Transferred over from an import? Randomly created? Entries that were wiped out; are you missing any?

**Reply by XRTce** [reply]
i am not sure where they came from. iam not missing any entries and i never imported into my vault. i also do not remember adding these entries. for debugging purposes i created another account and imported my vault into it. there i noticed that the 3 items had the name "---", i lokked in my original vault and the name was null

**Reply by XRTce** [reply]
i still have a encrypted json export which includes these entries. i imported that file to a test account. the entries are completely null. After import the name is set to "--" but no password, url or anything is set. maybe that helps

**Reply by BlackDex** [reply]
If you are willing to provide that file, or a snippet of the part that you think is causing the issue, that would be great.

**Reply by XRTce** [reply]
Sadly the original export file from my main vault is encrypted. But i imported the encrypted json into a test account and then exported the vault again as planm json. These are the two issue causing items. They are right at the start of the items array:
``` JSON
  "items": [
    {
      "passwordHistory": null,
      "revisionDate": "2024-09-19T12:26:10.803Z",
      "creationDate": "2024-09-19T12:26:10.803Z",
      "deletedDate": null,
      "id": "74f8f483-d891-4916-a3cf-085386aa2e03",
      "organizationId": null,
      "folderId": null,
      "type": 1,
      "reprompt": 0,
      "name": "--",
      "notes": null,
      "favorite": false,
      "login": {
        "fido2Credentials": [],
        "uris": [],
        "username": null,
        "password": null,
        "totp": null
      },
      "collectionIds": null
    },
    {
      "passwordHistory": null,
      "revisionDate": "2024-09-19T12:26:10.802Z",
      "creationDate": "2024-09-19T12:26:10.801Z",
      "deletedDate": null,
      "id": "e4bd83a1-7567-45e8-b1b7-c4d41f014adc",
      "organizationId": null,
      "folderId": null,
      "type": 1,
      "reprompt": 0,
      "name": "--",
      "notes": null,
      "favorite": false,
      "login": {
        "fido2Credentials": [],
        "uris": [],
        "username": null,
        "password": null,
        "totp": null
      },
      "collectionIds": null
    }]
```

Please notice that the name of these items not being null but "--" is because of the importing of the encrypted json. If there is a way i could decrypt the JSON without importing it i could do that and provide the original JSON Snippet

Hope I can help

**Reply by Gerardv514** [reply]
Did we ever determine what the issue was here?

**Comment by yry-dev** [comment]
I am getting this same error now after ios updates and I'm pretty sure DNS isn't the culprit as I can stil log into the vault from the web on safari.

**Reply by XRTce** [reply]
From my understanding this could be because at one place in your vault there is a null string which is not supported by the new Bitwarden app. You could do a unencrypted json export of your vault and search for ": null" and ": --" to find these values. Then you could try to remove these entries if not needed or fill them with a valid string and import the file into a new account. Do this until you got no error

**Comment by JohanEnberg** [comment]
I have the same issue. Can login in to web but iphone gives error. Exported vault to json. There are no "--" entries and non of the "name" fields are "null" but almost every entry has at least one "null" field. Is it only the name field that can't be "null" or is it any field?
